src/main.cpp

```diff
@@ -2354,7 +2363,7 @@ static DisconnectResult DisconnectBlock(const CBlock& block, CValidationState& s
     // move best block pointer to prevout block
     view.SetBestBlock(pindex->pprev->GetBlockHash());
 
-    if (fUpdateAddressIndex) {
+    if (addressIndex.size() > 0) {
```

Comment:
This change isn't needed, but this seems cleaner and clearer.

Reply:
Nit: I find the previous code easier to read and it matches the convention of checking global flags.
Nit: Comments to identify/segregate the explorer code

Reply:
> Nit: I find the previous code easier to read and it matches the convention of checking global flags.

Fixed: https://github.com/zcash/zcash/pull/3849/files#diff-7ec3c68a81efff79b6ca22ac1f1eabbaR2370